dynamicLibrary/CMakeLists.txt

```diff
@@ -74,9 +74,20 @@ set_target_properties(shastaDynamicLibrary PROPERTIES DEFINE_SYMBOL "")
 # Python 3 and pybind11.
 # I was not able to get find_package to work with pybind11 as installed by pip3.
 # The code below creates include paths for both Python 3 and pybind11.
+
+# Python 3.8 requires the `--embed` flag to be passed in.
+# find_python cmake module was introduced in v3.12, which isn't available for installation
+# via `apt` on Ubuntu 18.04 LTS. So I am not using it. 
+execute_process(COMMAND /usr/bin/python3-config --embed --libs OUTPUT_VARIABLE
```

Comment:
Would it be more straightforward to use `python3 --version`, then use its output to decide whether to add the `--embed` flag?

Reply:
Considered it. Decided against it because we don't know if this change is permanent. They might revert back for reasons we don't know of in a later version. If they do, then the conditional gets more complicated and we might have to maintain a white list of versions that require the `--embed` flag. It also made the code more complex. I couldn't do the recommended way because `find_python` does not work on the version of `cmake` available through `apt` on Ubuntu 18.04.

Eventually, as we drop support for older versions of Ubuntu, we can clean up this mess.